navit/maptool/tile.c

```diff
@@ -405,7 +405,10 @@ write_aux_tiles(struct zip_info *zip_info)
 		assert(buffer != NULL);
 		f=fopen(at->filename,"rb");
 		assert(f != NULL);
-		fread(buffer, at->size, 1, f);
+		
+	        if (fread(buffer, at->size, 1, f) == 0){
+			dbg(lvl_warning, "fread failed");
```

Comment:
we should probably skip to the next iteration if the read failed, not forgetting to close (f)